enhancements/machine-api/control-plane-set.md

```diff
@@ -0,0 +1,603 @@
+with the process outlined earlier in this document.
+
+#### Machine Config may change during a rollout
+
+There may be occasions where the Machine Config Operator attempts to rollout new Machine Config during a Control Plane
+scaling operation. We do not believe this will cause issue, but it may extend the time taken for the scaling operation
+to take place.
+
+While a scaling operation is in progress, etcd quorum is protected by the protection mechanism mentioned above as well
+as the etcd quorum guard. The quorum guard will ensure that at most one etcd instance is interrupted at any time.
+If the Machine Config operator needs to rollout an update, it will proceed in the usual manner while the etcd learner
+process (part of scaling up a new etcd member) may suffer a delay due to the restart of the Control Plane Machines.
+
+#### A user deletes the ControlPlaneSet resource
+
+Users will be allowed to remove ControlPlaneSet resources should they so desire. This shouldn't pose a risk to the
+control plane as the ControlPlaneSet will orphan Machines it manages before it is removed from the cluster. More detail
+is available in the [Removing/disabling the ControlPlaneSet](#Removingdisabling-the-ControlPlane) notes.
+
+#### The ControlPlaneSet spec may not match the existing Machines
+
+It is likely that in a number of scenarios, the spec attached to the ControlPlaneSet for creating new Machines may not
+match that of the existing Machines. In this case, we expect the ControlPlaneSet operator will attempt a rolling update
+when it is created. Apart from refreshing the Machines potentially needlessly, this shouldn't have a negative effect on
+the cluster assuming the new configuration is valid. If it is invalid, the cluster will degrade as the new Machines
+created by the ControlPlaneSet will fail to launch.
+
+We have designed this operator with IPI style clusters in mind. We expect that the Machine objects within the cluster
+should match the actual hosts on the infrastructure provider. In IPI clusters this is ensured by the installer.
+Users may have, out of band, then modified these instances, but unfortunately we have no way to deal with this at
+present. In these scenarios, when the Machines are replaced, they may not exactly match the previous iteration and
+therefore may cause issues for the customer.
+If a customer wishes to mitigate the chances of issues occurring, they may wish to manually control the roll out using
+the OnDelete update strategy.
+
+## Design Details
+
+### Open Questions
+
+Some open questions currently exist, but are called out in the Recreate Update Strategy notes.
+
+1. How do we wish to introduce this feature into the cluster, should it be a Technical Preview in the first instance?
+2. Do we want this to be an optional add-on (delivered via OLM) or do we want this to be a default operator for future
+   clusters?
```

Comment:
Is a distinct life-cycle from MCO and etcd operator even feasible?

Productizing things via cPaaS is still super complex. Would rather like to see the development time spent into making this mechanism rock-solid than wasting it in wrongly understood modularity of the platform.